check_pedigree.py

```diff
@@ -0,0 +1,1217 @@
+         
+            # for comparision purposes to ibd output, set sex and family to none (sex is already checked in a separate step)
+            Sex = None
+            Family_ID = None
+         
+            if Individual_ID in vcf_samples:
+                if Paternal_ID in vcf_samples and Maternal_ID in vcf_samples:
+                    ped_trios.append(hl.Trio(Individual_ID, Family_ID, Paternal_ID, Maternal_ID,Sex))
+                elif Paternal_ID in vcf_samples and Maternal_ID not in vcf_samples:
+                    ped_duos.append(sorted([Individual_ID,Paternal_ID]))
+                elif Paternal_ID not in vcf_samples and Maternal_ID in vcf_samples:
+                    ped_duos.append(sorted([Individual_ID,Maternal_ID]))
+                
+
+    given_trios = hl.Pedigree(ped_trios)
+    out_new_ped.close()
+    
+    return(given_trios, ped_duos, given_families, seqr_projects)
+
+
+
+def annotate_parent_decisions(
+    trios: hl.Pedigree,
+    duos: list,
+    ht: hl.Table
+) -> hl.Table:
+
+    """
+    Annotates who the parents are for trios and duos
+    :param input_pedigree: str
+    :param vcf_samples: dict
+    :return: ht with annotated decision columns for who the parent(s) of the sample were
+    :rtype: hl.Table
+    """
+    
+    # filter out confirmed siblings
+    parent_decisions = {}
+
+    for trio in trios:
+        parent_decisions[trio.s] = [trio.pat_id, trio.mat_id]  # two parents
+    
+    for duo in duos:
+        s1 = duo[0]
+        s2 = duo[1]
+        parent_decisions[s1] = [str(s2)]  # duos
+        parent_decisions[s2] = [str(s1)]  # duos
+    
+    
+    hl_final_decisions = hl.literal(parent_decisions)
+
+    
+    ht = ht.annotate(decision_i = hl_final_decisions.get(ht.i),
+                                    decision_j = hl_final_decisions.get(ht.j))
+
+
+    # set NAs to no decision
+    dec_exprj = hl.if_else(hl.is_missing(ht.decision_j), ["no_decision"],  ht.decision_j)
+    ht = ht.annotate(decision_j = dec_exprj)
+
+    dec_expri = hl.if_else(hl.is_missing(ht.decision_i), ["no_decision"],  ht.decision_i)
+    ht = ht.annotate(decision_i = dec_expri)
+    
+    return(ht)
+
+
+
+def evaluate_trios(
+    ht: hl.Table,
+    master_kin_ht: hl.Table,
+    out_summary: io.TextIOWrapper,
+    out_review: io.TextIOWrapper,
+    trios: hl.Pedigree,
+    given_trios: hl.Pedigree
+) -> Tuple[hl.Table, hl.Table, List[hl.Trio], hl.Trio, List[hl.Trio]]:
+
+    """
+    Determine if trios are confirmed, inferred only, or given only, annotate master kinship table with trio decisions
+    :param ht: hl.Table
+    :param master_ht: hl.Table
+    :return: ht with trio samples removed, master kinship ht with annotated trio decisions, given-only trios, complete inferred trios, inferred only trios
+    :rtype: Tuple[hl.Table, hl.Table, List[hl.Trio], hl.Trio, List[hl.Trio]]
+    """
+    
+    # set fam_id and is_female of inferred ped to none
```

Comment:
This first part seems rather complex to me -- all you want is an outer join of these two sets. Let's do this in hail

First, I'd suggest adding the following helper method to gnomad_methods.related. This will generally be useful.
```
def pedigree_to_ht(
    pedigree: hl.Pedigree
) -> hl.Table:
    return hl.Table.parallelize(
        hl.literal(
            [
                dict(
                    s=trio.s,
                    fam_id=trio.fam_id,
                    is_female=trio.is_female,
                    pat_id=trio.pat_id,
                    mat_id=trio.mat_id
                ) for trio in trios.trios
            ],
            hl.tarray(hl.tstruct(
                s=hl.tstr,
                fam_id=hl.tstr,
                is_female=hl.tbool,
                pat_id=hl.tstr,
                mat_id=hl.tstr)
            )
        )
    )
```
Now, with this helper, this all code becomes:
```
_trios = pedigree_to_ht(trios).select('s', 'pat_id', 'mat_id', inferred=True).key_by('s', 'pat_id', 'mat_id')
_trios = trios.join(
    pedigree_to_ht(given_trios).select('s', 'pat_id', 'mat_id', given=True).key_by('s', 'pat_id', 'mat_id'),
    how='outer
)
```

And then you can easily get what you want from there, maybe using something like:
```
classified_trios = _trios.aggregate(hl.agg.group_by([_trios.inferred, _trios.confirmed], hl.agg.collect())
```

Reply:
Doing this would also reduce the number of values returned as all these list would be combined into a single dict. I'd think about splitting this function into nuclear action that return a single output if at all possible. Something like:
- `get_inferred_vs_confirmed_trios`
- `annotate_master_kin_ht`
....